core/media_streamer.go

```diff
@@ -102,6 +103,34 @@ func (ms *mediaStreamer) NewStream(ctx context.Context, id string, reqFormat str
 	return s, nil
 }
 
+func (ms *mediaStreamer) TranscodeDownload(ctx context.Context, mf *model.MediaFile, format string, bitrate int, w io.Writer) error {
+	if bitrate == 0 {
+		bitrate = mf.BitRate
+	}
+
+	job := &streamJob{
+		ms:      ms,
+		mf:      mf,
+		format:  format,
+		bitRate: bitrate,
+	}
+
+	r, error := ms.cache.Get(ctx, job)
+
+	if error != nil {
+		return error
+	}
+
+	_, err := io.Copy(w, r)
```

Comment:
I think we need to close the reader returned by the `ms.Cache.Get()` method, like we do when calling `NewStreamer()`. By the way, why not use `NewStreamer` to obtain a `Reader` inside the `Archiver` and do the copy there? Why the need to introduce this `TranscodeDownload` method?